src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/GameManager.java

```diff
@@ -1095,8 +1122,8 @@ private GameboardItem augmentGameItemWithAttemptInformation(final GameboardItem
         gameItem.setQuestionPartStates(questionPartStates);
         Integer questionPartsTotal = questionPartsCorrect + questionPartsIncorrect + questionPartsNotAttempted;
         gameItem.setQuestionPartsTotal(questionPartsTotal);
-        Float percentCorrect = 100f * new Float(questionPartsCorrect) / questionPartsTotal;
-        Float percentIncorrect = 100f * new Float(questionPartsIncorrect) / questionPartsTotal;
+        Float percentCorrect = 100f * questionPartsCorrect / questionPartsTotal;
+        Float percentIncorrect = 100f * questionPartsIncorrect / questionPartsTotal;
```

Comment:
Does this definitely avoid the integer division issues?

Reply:
It does, although I agree, I was sceptical too. 
> *15.17. Multiplicative Operators*
> The operators *, /, and % are called the multiplicative operators.
> The multiplicative operators have the same precedence and are syntactically left-associative (they group left-to-right).
> ...
> If the promoted type is float or double, then floating-point arithmetic is performed.
> https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.17

It was IntelliJ that suggested the change... happy to rearrange it and add a cast so that it is more explicit if you like though:

`float percentCorrect = ((float)questionPartsCorrect / questionPartsTotal) * 100`